### PR TITLE
Show latest deployable version at the top of the version history page

### PR DESCRIPTION
**Pull request by sgalsaleh**

### CODE REVIEW

**Review by GraysonNull — APPROVED** [review]

File: pkg/handlers/app.go [hunk 1]
```diff
@@ -298,6 +298,7 @@ func (h *Handler) GetAppVersionHistory(w http.ResponseWriter, r *http.Request) {
 	pageSize := 20
 	currentPage := 0
 	pinLatest, _ := strconv.ParseBool(r.URL.Query().Get("pinLatest"))
+	pinLatestDeployable, _ := strconv.ParseBool(r.URL.Query().Get("pinLatestDeployable"))
```
Comment (sgalsaleh) [inline pkg/handlers/app.go:301]: new

File: pkg/store/kotsstore/downstream_store.go [hunk 2]
```diff
@@ -359,19 +359,30 @@ func (s *KOTSStore) FindDownstreamVersions(appID string, downloadedOnly bool) (*
...
-func (s *KOTSStore) GetDownstreamVersionHistory(appID string, clusterID string, currentPage int, pageSize int, pinLatest bool) ([]*downstreamtypes.DownstreamVersion, error) {
+func (s *KOTSStore) GetDownstreamVersionHistory(appID string, clusterID string, currentPage int, pageSize int, pinLatest bool, pinLatestDeployable bool) ([]*downstreamtypes.DownstreamVersion, error) {
 	versions, err := s.GetDownstreamVersions(appID, clusterID, false)
 	if err != nil {
 		return nil, errors.Wrapf(err, "failed to get downstream versions without details")
 	}
 
+	desiredVersions := []*downstreamtypes.DownstreamVersion{}
+
+	if pinLatestDeployable {
```
Comment (sgalsaleh) [inline pkg/store/kotsstore/downstream_store.go:370]: new

File: pkg/store/kotsstore/downstream_store.go [hunk 3]
```diff
@@ -359,24 +359,33 @@ func (s *KOTSStore) FindDownstreamVersions(appID string, downloadedOnly bool) (*
...
-func (s *KOTSStore) GetDownstreamVersionHistory(appID string, clusterID string, currentPage int, pageSize int, pinLatest bool) ([]*downstreamtypes.DownstreamVersion, error) {
+func (s *KOTSStore) GetDownstreamVersionHistory(appID string, clusterID string, currentPage int, pageSize int, pinLatest bool, pinLatestDeployable bool) ([]*downstreamtypes.DownstreamVersion, error) {
 	versions, err := s.GetDownstreamVersions(appID, clusterID, false)
 	if err != nil {
 		return nil, errors.Wrapf(err, "failed to get downstream versions without details")
 	}
 
-	startIndex := currentPage * pageSize
-	endIndex := currentPage*pageSize + pageSize
+	desiredVersions := []*downstreamtypes.DownstreamVersion{}
```
Comment (sgalsaleh) [inline pkg/store/kotsstore/downstream_store.go:368]: new

File: pkg/store/kotsstore/downstream_store.go [hunk 4]
```diff
@@ -359,24 +359,33 @@ func (s *KOTSStore) FindDownstreamVersions(appID string, downloadedOnly bool) (*
...
-	desiredVersions := []*downstreamtypes.DownstreamVersion{}
-	for i, v := range versions.AllVersions {
-		if pinLatest && i == 0 {
-			desiredVersions = append(desiredVersions, v)
-			continue
+
+	if pinLatestDeployable {
+		latestDeployableVersion, _, _, err := s.getLatestDeployableDownstreamVersion(appID, clusterID, versions)
+		if err != nil {
+			return nil, errors.Wrapf(err, "failed to get next downstream version")
```
Comment (emosbaugh) [inline pkg/store/kotsstore/downstream_store.go:379]:
```suggestion
			return nil, errors.Wrapf(err, "failed to get latest deployable downstream version")
```

File: pkg/store/kotsstore/downstream_store.go [hunk 5]
```diff
@@ -359,24 +359,33 @@ func (s *KOTSStore) FindDownstreamVersions(appID string, downloadedOnly bool) (*
...
+			desiredVersions = append(desiredVersions, versions.AllVersions[0])
+		}
 	}
-	desiredVersions := []*downstreamtypes.DownstreamVersion{}
-	for i, v := range versions.AllVersions {
-		if pinLatest && i == 0 {
-			desiredVersions = append(desiredVersions, v)
-			continue
+
+	if pinLatestDeployable {
```
Comment (emosbaugh) [inline pkg/store/kotsstore/downstream_store.go:376]: so latest and latest deployable will always be included as first for ALL pages and page size will always be desired page size + 2 if both flags are enabled. is this intended?

Comment (sgalsaleh) [reply pkg/store/kotsstore/downstream_store.go:376]: yes

**Review by emosbaugh — APPROVED** [review]

**Pull request merged**